packages/dev/core/src/Collisions/gpuPicker.ts

```diff
@@ -122,20 +160,14 @@ export class GPUPicker {
     private _generateColorData(instanceCount: number, id: number, index: number, r: number, g: number, b: number, onInstance: (i: number, id: number) => void) {
         const colorData = new Float32Array(4 * (instanceCount + 1));
 
-        colorData[0] = r / 255.0;
-        colorData[1] = g / 255.0;
-        colorData[2] = b / 255.0;
-        colorData[3] = 1.0;
+        GPUPicker._SetColorData(colorData, 0, r, g, b);
+
         for (let i = 0; i < instanceCount; i++) {
-            const r = (id & 0xff0000) >> 16;
-            const g = (id & 0x00ff00) >> 8;
-            const b = (id & 0x0000ff) >> 0;
+            const { r, g, b } = GPUPicker._IdToRgb(id);

```

Comment:
The extra GC issue would be solved with a temp object.

Reply:
Changed.